app/models/team.rb

```diff
@@ -63,10 +63,21 @@ def in_current_season
       where(season: Season.current)
     end
 
+    def by_season_year(year)
```

Comment:
Idea: making this `by_season(year_or_season)` that automatically detects whether the argument is a real `Season` object or a year (either as an integer or a string) and then constructs the `where` condition accordingly.